src/Features/LanguageServer/Protocol/Extensions/Extensions.cs

```diff
@@ -34,6 +40,13 @@ public static ImmutableArray<DocumentId> GetDocumentIds(this Solution solution,
             return GetDocuments<DocumentId>(solution, documentUri, (s, i) => i);
         }
 
+        public static ImmutableArray<Document> GetDocuments(this Solution solution, Uri documentUri, string? clientName)
+        {
+            var documents = GetDocuments<Document>(solution, documentUri, (s, i) => s.GetRequiredDocument(i));
```

Comment:
Is there a reason the GetDocuments() call just doesn't return ImmutableArray<Document> directly as opposed to making everybody do a bunch of lambdas? This is otherwise equivalent to the one two methods above.

Reply:
Yeah, I'm not a fan of any of these extension methods that all take lambdas that call each other. I only looked at cleaning up the solution provider ones, but I can have a pass at the rest. I think this is part of an existing one that I pulled the filtering code out of, but the diff of this file looks pretty terrible in general.

Reply:
I cleaned all of this stuff up.